File: Server/DiscussionService/DiscussionServiceDomain/Models/Discussion.cs
namespace DiscussionServiceDomain.Models
{
    public class Discussion : IdModel
    {
        public string Title { get; private set; } = string.Empty;
        public string Description { get; private set; } = string.Empty;
        public DateTime StartAt { get; private set; } = DateTime.UtcNow;
        public Guid CreatedBy { get; private set; } 
        public ICollection<Guid> Subscribers { get; private set; } = [];
        public bool IsActive { get; set; } = false;

        public Discussion() { }

        public Discussion(string title, string description, DateTime startAt, Guid createdBy, bool isActive, ICollection<Guid> subscribers)
        {
            Title = title;
            Description = description;
            StartAt = startAt;
            CreatedBy = createdBy;
            IsActive = isActive;
            Subscribers = subscribers;
        }
    }
}

File: Client/reactclient/src/types/discussion.ts
export interface Discussion {
  id: string;
  title: string;
  description: string;
  startAt: string;
  createdBy: string;
  isActive: boolean;
  subscribers: string[];
}

export interface CreateDiscussionRequest {
  createdBy: string | null;
  title: string;
  description: string;
  startAt: string;
}

export interface UpdateDiscussionRequest {
  id: string;
  updatedBy: string | null;
  title: string;
  description: string;
}

export interface Message {
  userId: string;
  text: string;
  username: string;
  sentAt: string;
}
